src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -509,14 +514,35 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       callOutputs foreach updateSymbolCache(callKey)
       logger.debug(s"In state WorkflowRunning: Got PersistenceCompleted message for Done call ${callKey.tag}")
       val updatedData = startRunnableCalls(data)
-      if (isWorkflowDone) scheduleTransition(WorkflowSucceeded)
       val finalData = updatedData.removePersisting(callKey, ExecutionStatus.Done)
+      self ! CheckForWorkflowComplete
       stay using finalData
     case Event(PersistenceSucceeded(callKey, ExecutionStatus.Preempted, callOutputs), data) =>
       executionStore += callKey -> ExecutionStatus.Preempted
       val updatedData = startRunnableCalls(data)
       val finalData = updatedData.removePersisting(callKey, ExecutionStatus.Preempted)
+      self ! CheckForWorkflowComplete
       stay using finalData
+    case Event(PersistenceSucceeded(callKey, ExecutionStatus.Failed, callOutputs), data) =>
+      executionStore += callKey -> ExecutionStatus.Failed
+      val finalData = data.removePersisting(callKey, ExecutionStatus.Failed)
+      self ! CheckForWorkflowComplete
+      stay using finalData
+    case Event(CheckForWorkflowComplete, data) =>
+      checkForWorkflowComplete(data)
+      stay()
+  }
+
+  private def checkForWorkflowComplete(data: WorkflowData): Unit = {
+    val executionStatuses = executionStore.values
+
+    if (!(executionStatuses exists ( _ == ExecutionStatus.Running)) && data.pendingExecutions.isEmpty) {
```

Comment:
Since it's an Iterable, it doesn't have contains. I could "toSeq" it but that seems like slight overkill